admin can destroy events and organization can cancel events

Omarkojak opened the pull request:
for issue #134

Omarkojak committed:
#134 | Now admin can destroy events and organization can cancel events

Omarkojak commented:
@AhmadElsagheer

AhmadElsagheer commented on app/Http/Controllers/Event/EventController.php:
```diff
@@ -109,11 +111,20 @@ public function update(EventRequest $request, $id)
+	 * Organization cancel an event
+	 */
+	public function cancel($id)
+	{
```
You must check here that the authenticated organization is the one who created the event

Omarkojak committed:
Resolving conflicts

# Conflicts:
#	app/Http/Controllers/Event/EventController.php
#	app/Http/Services/EventService.php

Omarkojak committed:
#134 | checking whether this event is for this organization

Omarkojak commented:
@AhmadElsagheer

AhmadElsagheer commented on app/Http/Controllers/Event/EventController.php:
```diff
@@ -36,9 +36,11 @@ public function __construct()
         ]]);
-    }
+
+		$this->middleware('auth_admin', ['only' => ['destroy']]);
```
fix identation

Omarkojak replied:
ana 3ndy mzbota msh 3arf leh :D

AhmadElsagheer commented on app/Http/Controllers/Event/EventController.php:
```diff
@@ -116,12 +118,24 @@ public function update(EventRequest $request, $id)
 
 	/**
-	 * Cancel an event.
-	 * @param int $id event id.
```
add this line below the `Organization cancel an event`

AhmadElsagheer commented on app/Http/Controllers/Event/EventController.php:
```diff
@@ -116,12 +118,24 @@ public function update(EventRequest $request, $id)
+	{
+		if(auth()->guard('organization')->user()->id == Event::findorfail($id)->organization()->id)
+		{
+			$this->eventService->cancel($id);
```
Pass the organization to the cancel function as this function is used by both web and API. Checking for the authenticated organization must be done in the service. The service must return true or false.

AhmadElsagheer commented on app/Http/Services/EventService.php:
```diff
@@ -56,15 +56,24 @@ public function update(EventRequest $request, $id)
+
+	/**
+	 * Delete an event.
+	 */
```
Add `@param` to indicate what this `id` is

AhmadElsagheer commented on app/Http/Services/EventService.php:
```diff
@@ -56,15 +56,24 @@ public function update(EventRequest $request, $id)
+	{
+		$event = Event::findOrFail($id);
+		$event->forceDelete();
+		Notification::notify($event->volunteers, 3, null,
```
You need to send the notification and unindex the event before deletion?

Omarkojak replied:
maybe i don't need to unindex but i need to notify and that was stated in the issue

Omarkojak replied:
i tested it and we needed unindexing the event too

AhmadElsagheer replied:
I mean do you need it before or after the delete call? Is it working fine?

AhmadElsagheer commented on app/Http/routes.php:
```diff
@@ -251,6 +251,11 @@
 
+    /*
+     * Event Cancelling
+     */
```
remove doc and add it below CRUD doc but above the resource

AhmadElsagheer commented:
Need to handle `cancel` function only in `EventAPIController`

Omarkojak committed:
#134 | Now admin can destroy events and organization can cancel events

Omarkojak committed:
#134 | API handling

Omarkojak commented:
@AhmadElsagheer

Omarkojak committed:
Removing the try and catch and using volunteer parameter in register function
